web-app/js/portal/filter/BooleanFilter.js

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+Ext.namespace('Portal.filter');
+
+Portal.filter.BooleanFilter = Ext.extend(Portal.filter.Filter, {
+
+    getSupportedGeoserverTypes: function() {
+
+        return ['boolean'];
```

Comment:
Yep, I think I like option 2 more, much better from an encapsulation point of view. At the end of the day, the configuration for supported geoserver types is still in one location per filter type here if changes need to be made.